### PR TITLE
[WFLY-18952] Migrate operations for MP OpenTracing and MP Metrics are failing

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-18952

Correct the path handling code in the migrate ops
Fix typo on extension name for MP Metrics

### CODE REVIEW

File: legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java [hunk 1]
```diff
@@ -167,14 +167,18 @@ private void addOpenTelemetryExtension(final OperationContext context,
 
     private void addOpenTelemetrySubsystem(final OperationContext context,
                                            final Map<PathAddress, ModelNode> migrateOperations) {
+        PathAddress subsystemAddress = context.getCurrentAddress()
```
Comment (bstansberry) [inline legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java:170]: PathAddress parentAddress = context.getCurrentAddress().getParent();

File: legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java [hunk 2]
```diff
@@ -167,14 +167,18 @@ private void addOpenTelemetryExtension(final OperationContext context,
...
+                .getParent()
+                .append(OPENTELEMETRY_SUBSYSTEM_ELEMENT);
         Resource root = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS, false);
```
Comment (bstansberry) [inline legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java:173]: Resource root = context.readResourceFromRoot(parentAddress, false);

File: legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java [hunk 3]
```diff
@@ -167,14 +167,18 @@ private void addOpenTelemetryExtension(final OperationContext context,
...
-        if (root.hasChild(OPENTELEMETRY_SUBSYSTEM_ELEMENT)) {
+        try {
+            root.navigate(subsystemAddress);
```
Comment (bstansberry) [inline legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java:175]: Remove new lines 174, 175, 177; restore old line 171

File: legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java [hunk 4]
```diff
@@ -167,14 +167,18 @@ private void addOpenTelemetryExtension(final OperationContext context,
...
             return; // subsystem is already added, nothing to do
+        } catch (Resource.NoSuchResourceException e) {
         }
 
-        PathAddress address =  pathAddress(OPENTELEMETRY_SUBSYSTEM_ELEMENT);
```
Comment (bstansberry) [inline legacy/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/MigrateOperation.java:175]: PathAddress subsystemAddress = parentAddress.append(OPENTELEMETRY_SUBSYSTEM_ELEMENT);

**Review by bstansberry — CHANGES_REQUESTED** [review]
This won't work as readResourceFromRoot(address, false) will return a resource that has no grandchildren (the javadoc explains the details).  So root.navigate will not work properly in a domain, where the subsystem resource is a grandchild of the root, not a child.

My specific comments below will fix it.

**Comment by jasondlee** [comment]
> My specific comments below will fix it.

Changes made and pushed.

**Review by emmartins — APPROVED** [review]
Tested manually with the migration tool (cherry-picking this commit to xp5 dev branch), all good, both ops did run successfully, no sign of the reported issues.

**Pull request merged**